src/components/TodoItem/TodoItem.tsx

```diff
@@ -0,0 +1,119 @@
+import classNames from 'classnames';
+import { useState } from 'react';
+import { Todo } from '../../types/Todo';
+
+type Props = {
+  todo: Todo;
+  todosLoadingState: Todo[],
+  onRemoveTodo: (todo: Todo) => void;
+  onToogleUpdateTodo: (todo: Todo) => void;
+  onHandleUpdate: (todo: Todo) => void;
+};
+
+export const TodoItem: React.FC<Props> = ({
+  todo,
+  todosLoadingState,
+  onRemoveTodo,
+  onToogleUpdateTodo,
+  onHandleUpdate,
+}) => {
+  const { title, completed } = todo;
+  const [isEditing, setIsEditing] = useState(false);
+  const [todoTitle, setTodoTitle] = useState(title);
+
+  const eventChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    const { value } = e.target;
+
+    setTodoTitle(value);
+  };
+
+  const handleOnSubmit = (e: React.FormEvent) => {
+    e.preventDefault();
+    setIsEditing(false);
+
+    const trimTodoTitle = todoTitle.trim();
+
+    if (trimTodoTitle === todo.title) {
+      return setTodoTitle(title);
+    }
+
+    if (!trimTodoTitle) {
+      return onRemoveTodo(todo);
+    }
+
+    return onHandleUpdate({
+      ...todo,
+      title: todoTitle,
+    });
+  };
+
+  const handleCancelEditing = (
+    event: React.KeyboardEvent<HTMLInputElement>,
+  ) => {
+    if (event.code === 'Escape') {
+      setIsEditing(false);
+      setTodoTitle(todo.title);
+    }
+  };
+
+  const hasLoadingState = todosLoadingState
+    .some(todoLoading => todoLoading.id === todo.id);
+
+  const isLoading = todo.id === 0 || hasLoadingState;
+
+  return (
+    <li
+      className={classNames(
+        'todo',
+        { completed },
+      )}
+      onDoubleClick={() => setIsEditing(true)}
```

Comment:
the newly appeared input should be focused
and move this to a separate function